services/src/main/groovy/org/jfrog/artifactory/client/impl/ReplicationsImpl.groovy

```diff
@@ -0,0 +1,156 @@
+        }
+    }
+
+    @Override
+    void createOrReplaceReplication(Replication replication) {
+        artifactory.put("${getReplicationsApi()}${repoKey}", [:], ContentType.ANY, null, ContentType.JSON, replication)
+    }
+
+    @Override
+    void createOrReplaceReplications(Collection<LocalReplication> replications) {
+        assert replications
+
+        // Determine the type of the repository (not all repository types support replications)
+        def repository = artifactory.repository(repoKey).get()
+
+        if (!repository) {
+            throw new RuntimeException("The repository '${repoKey}' doesn't exist")
+        }
+
+        def path = "${getReplicationsMultipleApi()}${repoKey}"
+
+        // The code below adapts a 'Get Repository Replication Configuration' service response into a
+        // 'Create or Replace Local Multi-push Replication' service request
+
+        // Use the first replication to infer the shared properties
+        def first = replications.iterator().next() as LocalReplication
+
+        // Convert the incoming list of replications into the correct request
+        def array = new JSONArray()
+
+        replications.each { replication ->
+            def object = new JSONObject()
+            object.put('url', replication.url)
+            object.put('socketTimeoutMillis', replication.socketTimeoutMillis)
+            object.put('username', replication.username)
+            object.put('password', replication.password)
+            object.put('enableEventReplication', replication.enableEventReplication)
+            object.put('enabled', replication.enabled)
+            object.put('syncDeletes', replication.syncDeletes)
+            object.put('syncProperties', replication.syncProperties)
+            object.put('syncStatistics', replication.syncStatistics)
+            object.put('repoKey', replication.repoKey)
+
+            array.add(object)
+        }
+
+        def payload = new JSONObject()
+        payload.put('cronExp', first.cronExp)
+        payload.put('enableEventReplication', first.enableEventReplication)
+        payload.put('replications', array)
+
+        if (repository.rclass == RepositoryTypeImpl.LOCAL) {
+            artifactory.put(path, [:], ContentType.ANY, null, ContentType.JSON, payload)
+        } else {
+            throw new UnsupportedOperationException("The method isn't supported for a ${repository.rclass} repository")
+        }
+    }
+
+    @Override
+    void delete() {
+        // Determine the type of the repository (not all repository types support replications)
+        def repository = artifactory.repository(repoKey).get()
+
+        if (!repository) {
+            throw new RuntimeException("The repository '${repoKey}' doesn't exist")
+        }
+
+        def path = "${getReplicationsApi()}${repoKey}"
+
+        if ((repository.rclass == RepositoryTypeImpl.LOCAL) || (repository.rclass == RepositoryTypeImpl.REMOTE)) {
```

Comment:
As mentioned earlier, this if statement can be probably simplified to != RepositoryTypeImpl.VIRTUAL